packages/grafana-o11y-ds-frontend/src/TemporaryAlert.test.tsx

```diff
@@ -0,0 +1,19 @@
+import { act, render, screen } from '@testing-library/react';
+import React from 'react';
+
+import { TemporaryAlert } from './TemporaryAlert';
+
+describe('TemporaryAlert', () => {
+  it('full component life cycle', async () => {
+    render(<TemporaryAlert duration={1000} severity="error" text="" />);
+    expect(screen.queryByTestId('data-testid Alert error')).not.toBeInTheDocument();
+
+    render(<TemporaryAlert duration={1000} severity="error" text="Error message" />);
+    expect(screen.queryByTestId('data-testid Alert error')).toBeInTheDocument();
+    expect(screen.getByText('Error message')).toBeInTheDocument();
+
+    await act(() => new Promise((_) => setTimeout(_, 1000)));
```

Comment:
Instead of this, you can use ` jest.useFakeTimers()` to handle timeouts. And if you added `duration` just for tests, then using ` jest.useFakeTimers()` can remove the need for it. 🙂